src/evict/evict_lru.c

```diff
@@ -404,7 +399,13 @@ __evict_server(WT_SESSION_IMPL *session, bool *did_work)
 		if ((ret = __evict_lock_handle_list(session)) == EBUSY)
 			return (0);
 		WT_RET(ret);
+
+		/*
+		 * Clear the walks so we don't pin pages while asleep,
+		 * otherwise we can block applications evicting large pages.
+		 */
```

Comment:
I think eviction is one of the areas of the system that's been reworked so many times it tends to suffer from fragmentation. (Pretty much everybody in the WiredTiger team has worked on it more than a few times.)

Anyway, any clarity you can add to it would be welcome -- I'd suggest a new ticket though, and @agorrod as your reviewer.